cpp/ql/test/library-tests/ir/range-analysis/SimpleRangeAnalysis_tests.cpp

```diff
@@ -1028,4 +1028,75 @@ void test_negate_signed(int s) {
   if(10 < s && s < 20) {
     range<int>(-s); // $ range=<=-11 range=>=-19
   }
-}
\ No newline at end of file
+}
+
+// By setting the guard after the use in another guard we 
+// don't get the useful information
+void test_guard_after_use(int pos, int size, int offset) {
+  if (pos + offset >= size) { // $ overflow=+-
+    return;
+  }
+  if (offset == 1) {
```

Comment:
I guess this should be `!=` for the missing range to make sense, right?
```suggestion
  if (offset != 1) {
```

Reply:
Good catch. Yes, I think so. That's what we saw in the original example as well: https://github.com/scummvm/scummvm/blob/79c30639793777a465dd57023fbb60b82093e144/engines/composer/scripting.cpp#L206